tools/firmware.toit

```diff
@@ -577,14 +577,26 @@ find_esptool_ -> List:
   if bin_name.ends_with ".toit":
     if dir == "": dir = "."
     esptool_py := "$dir/../third_party/esp-idf/components/esptool_py/esptool/esptool.py"
-    if not file.is_file esptool_py:
-      throw "cannot find esptool in '$esptool_py'"
-    return ["python", esptool_py ]
+    if file.is_file esptool_py:
+      return ["python", esptool_py ]
   else:
     esptool := ["$dir/esptool$bin_extension"]
-    if not file.is_file esptool[0]:
-      throw "cannot find esptool in '$esptool[0]'"
+    if file.is_file esptool[0]:
+      return esptool
+  // Try to find esptool in PATH.
+  esptool := ["esptool$bin_extension"]
+  catch:
+    pipe.backticks esptool "version"
+    // Succeeded, so just return it.
     return esptool
+  // An exception was thrown.
+  // Try to find esptool.py in PATH.
+  if platform != PLATFORM_WINDOWS:
```

Comment:
Don't we need something similar on Windows?

Reply:
I don't thing shebang scripts work on Windows.
Python adds a file association of Python to the launcher, but I don't think that triggers when directly called (even through a shell).
I think on Windows we need a real executable.